pkg/chunk/chunk_store.go

```diff
@@ -65,6 +65,9 @@ type StoreConfig struct {
 
 	// When DisableIndexDeduplication is true and chunk is already there in cache, only index would be written to the store and not chunk.
 	DisableIndexDeduplication bool `yaml:"-"`
+
+	// This let's user mention the labels it wants to exclude while indexing as well as querying
+	ExcludeLabels util.ExcludeLabels `yaml:"exclude_labels" doc:"hidden"`
```

Comment:
Please register the CLI flag for this in `StoreConfig.RegisterFlags()`. Since you're using a custom data structure, you may need to define the CLI flag interface for serialization/deserialization.

Reply:
@pracucci Thanks. I have a small doubt. How do I register a CLI flag for a string? The `12768 ` is userID in the following and is a key in itself. How do I register this type of variable?
`exclude_labels:
    12768:
    - metric_name: go_gc_duration_seconds
      label_name: instance
    - metric_name: go_heap_size
      label_name: job
`

Reply:
Unfortunately, your data structure is not CLI flag friendly.  You would have to define your own custom format and then parse it, but I agree this data structure looks quite complicate to be represented in a CLI flag.

Reply:
Can you point me to an example of a custom format? I would sincerely appreciate it if I could use an example. Thanks in advance.

Reply:
Definitely! Look at `DurationList` defined in `pkg/storage/tsdb/config.go`.

Reply:
Thanks! 👍

Reply:
I think it's unreasonable to parse this structure as a CLI flag.

Reply:
I agree! It is getting extremely complicated.

Reply:
I agree it's complicated.

> doc:"hidden"

This will hide this config from the auto-generated config file reference because it expect a CLI flag for each config option. Just a fact to be aware.

Reply:
The lint error on CI still persists even when I use `doc:"hidden"`